### ISSUE TITLE
Add rules for this, delete, auto, lambda...

### ISSUE THREAD

**Issue by LuisGP** [richiware/cppstyle#5]
I want to call for discussing:

### this

Applying the already defined rules, I don't see the need for using `this` under any circumstance, so I propose to forbid its use. Using it will hide the break of the code stylish defined here.

### delete

To unify the usage of the `delete` operator I propose to use it always without parenthesis.

```cpp
delete(pointer_); // Bad
delete pointer_; // Good
```

### auto

I propose to define the rules for the usage of `auto` if any, and when it should be replaced by a concrete type, a `typedef` or a `using`.

### typedef vs. using

If you are going to define an alias for an already defined type, then `typedef` is your keyword, but when you need to simplify a type to make the code more readable in a well-defined scope, then which keyword use and why?

### lambda

Should we define a maximum length for lambda contents?

**Comment by lemunozm** [comment]
I agree with all changes, but if we allow the use of `typedef` and `using` as alias I will force to very restrictive cases (only when their usage really improve the readability). Do not allow to take free decisions to this.

For **typedef vs. using** discussion, read:
https://stackoverflow.com/questions/10747810/what-is-the-difference-between-typedef-and-using-in-c11

It appears that `using` is the winner (?)

**Comment by LuisGP** [comment]
So, they mean exactly the same. 
`using` allows templates and is more readable in general.

My vote is going to `using` then.
